From 0bd101db1c2193bbe9cd2ecaab25a75228799456 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Fri, 15 Jan 2016 03:34:52 -0500
Subject: [PATCH 2/4] wr-policy-kernel_t-mls-trusted-for-setting-process-level

kernel_t MLS trusted for setting the level of processes it executes.

This fixes avc denial issue:

  avc: denied { dyntransition } for  pid=1 comm="systemd" \
  scontext=system_u:system_r:kernel_t:s15:c0.c1023 \
  tcontext=system_u:system_r:init_t:s0-s15:c0.c1023 \
  tclass=process permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/kernel/kernel.te | 1 +
 1 file changed, 1 insertion(+)

diff --git a/policy/modules/kernel/kernel.te b/policy/modules/kernel/kernel.te
index 1b97a61..18f336f 100644
--- a/policy/modules/kernel/kernel.te
+++ b/policy/modules/kernel/kernel.te
@@ -295,6 +295,7 @@ mls_file_write_all_levels(kernel_t)
 mls_file_read_all_levels(kernel_t)
 mls_socket_write_all_levels(kernel_t)
 mls_fd_use_all_levels(kernel_t)
+mls_process_set_level(kernel_t)
 
 ifdef(`distro_redhat',`
 	# Bugzilla 222337
-- 
1.9.1

